main.c

```diff
@@ -239,6 +239,7 @@ static void can_msg_handler(const can_msg_t *msg) {
     // declare this in advance cause we can't declare it inside the switch
     uint8_t alt_num = 0;
     enum ARM_STATE desired_arm_state = ARMED;
+    int dest_id = get_reset_board_id(msg); // the destination board id for reset
```

Comment:
This needs to move down to line ~263, we only want to do this if the message is a reset message. The checks in get_reset_board_id should save us from any particular trouble, but we don't want to slow down the handler like this for every message.

Reply:
as for not being able to declare variable in a switch case, which is presumably why you did it this way.
you can either:
- just put get_reset_board_id() in the if statement, you don't need a variable (my preferred approach)

- or you can use {} to allow you to declare a variable as per this post: https://stackoverflow.com/questions/92396/why-cant-variables-be-declared-in-a-switch-statement